packages/evm/contracts/PermissionChecker.sol

```diff
@@ -256,7 +260,7 @@ abstract contract PermissionChecker is Core, Periphery {
         bytes calldata data,
         ParameterConfig memory parameter,
         ParameterPayload memory payload
-    ) private pure returns (Status, Trace[] memory nothing) {
+    ) private pure returns (Status, Trace[] memory nothing) { // TODO: weird to name this `nothing`. let's rename to `trace` here and destruct to it, while returning `trace()` instead of `nothing`?
         for (uint256 i; i < parameter.children.length; ) {
```

Comment:
Because it might have been assigned in previous iterations. In this case we return a fresh one